mlflow/transformers/__init__.py

```diff
@@ -654,11 +638,17 @@ def save_model(
     if conda_env is None:
         if pip_requirements is None:
             default_reqs = get_default_pip_requirements(built_pipeline.model)
-            # Infer the pip requirements with a timeout to avoid hanging indefinitely at prediction
-            inferred_reqs = infer_pip_requirements_with_timeout(
-                str(path), FLAVOR_NAME, fallback=default_reqs
-            )
-            default_reqs = sorted(set(inferred_reqs).union(default_reqs))
+            # NB: Skip inferring requirements for PEFT models to avoid loading the full pretrained
+            # model into memory. PEFT is mainly designed for fine-tuning large models under limited
+            # computational resources, so loading the full model is not preferred and can even crash
+            # the process due to OOM.
+            if not is_peft_model(built_pipeline.model):
+                # Infer the pip requirements with a timeout to avoid hanging at prediction
+                inferred_reqs = infer_pip_requirements_with_timeout(
+                    str(path), FLAVOR_NAME, fallback=default_reqs
+                )
+                default_reqs = set(inferred_reqs).union(default_reqs)
+            default_reqs = sorted(default_reqs)
```

Comment:
I didn't add any warning here because whether or not loading model or not during logging is our implementation detail for requirement logging, not indication of any errors. Similarly to why we use different logic depending on whether or not input_example is provided, but we don't say "hey we don't run prediction because you don't provide input_example". It's worth to show warning when fallback due to encountering any issues during inference, but here it is just our logic decide not to load the model. WDYT?

Reply:
Agreed. Let's not warn here as it's a conscious decision by the user to enable this serialization mode when using PEFT.